README.md

```diff
@@ -62,15 +61,12 @@ The reference's location and format depends on the following scenarios:
         * servicebinding/secret/host: {.status.address}
         * servicebinding/secret/`<binding_property>`: {.status.`<status_property>}` (where `<binding_property>` is any property from the binding schema, and `<status_property>` refers to the path to the correspoding `status` property)
 
-3. Regular k8s Deployment (Ingress, Route, Service, etc)  - An annotation in the corresponding CR that maps the `status` properties to their corresponding binding data. The value of this annotation can be specified in either [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) or [GO templates](https://golang.org/pkg/text/template/):
+3. Regular k8s Deployment (Ingress, Route, Service, Secret, ConfigMap etc)  - An annotation in the corresponding CR that maps the `status`, `spec` or `data` properties to their corresponding [binding data](#service-binding-schema). The value of this annotation can be specified in either [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) or [GO templates](https://golang.org/pkg/text/template/):
       * servicebinding/secret/host: {.status.ingress.host}
       * servicebinding/secret/host: {.status.address}
-      * servicebinding/secret/`<binding_property>`: status.`<status_property>` (where `<binding_property>` is any property from the binding schema, and `<status_property>` refers to the path to the correspoding `status` property)
+      * servicebinding/secret/`<binding_property>`: `<property_path>` (where `<binding_property>` is any property from the binding schema, and `<property_path>` refers to the path to the correspoding `status`, `spec` or `data` property)
 
-4. External service - An annotation in the local ConfigMap or Secret that bridges the external service.
-    * The annotation is in the form of either:
-      * servicebinding/configMap: self
-      * servicebinding/secret: self
+The above pattern can be used to expose external services (such as from a VM or external cluster), as long as there is an entity such as a Secret that provides the binding details. 
 
 ### Service Binding Schema
```

Comment:
Just thinking out loud about the whole idea of defining a schema... I agree that there are benefits in having a defined structure in the final binding secret such as users would know exactly what keys to expect in the secret.

However, with supporting more than one backing service per Service Binding Request, it becomes harder to define a single schema. For example, users might define a single service binding request for their entire app, and add kafka and db into a single ServiceBinding CR which would mean more than one `type`/`uri`/`password`.... It would be confusing for them to know which key-value pair is used for each service.

Having one backing service per ServiceBinding CR, makes it easier to define a schema but it soon becomes tricky with more backing services which might make it confusing for users.

I wonder how Cloud Foundry solved this issue

Reply:
ya, that's a fair point.  I agree that all apps which are consuming more than 1 service would need to add some prefixes to distinguish between the keys in their code.  I guess the value of having a proposed set of schema is to try to bring consistent:  so instead of address, host, endpoint, target, etc, we would consistently have "host".   It may have a prefix like "db2_host", "kafka_host", but at least it would be somewhat consistent. 

In the [buildpack spec](https://github.com/buildpacks/rfcs/blob/master/text/0012-service-binding.md) they went the same route as us (actually we copied them :) ) in terms of only defining mounted paths - which then leaves outside of the spec how application distinguish between the various keys that are mounted.  From the spec's perspective they all have unique paths they are mounted to - it's then up to the application how they consume that.